components/create_comment/create_comment.jsx

```diff
@@ -209,6 +212,24 @@ export default class CreateComment extends React.PureComponent {
         }
     }
 
+    focusTextboxIfNecessary = (e) => {
+        // Should only focus if RHS is expanded
+        if (!this.props.rhsExpanded) {
+            return;
+        }
+
+        // Bit of a hack to not steal focus from the channel switch modal if it's open
+        // This is a special case as the channel switch modal does not enforce focus like
+        // most modals do
+        if (document.getElementsByClassName('channel-switch-modal').length) {
+            return;
+        }
```

Comment:
I was originally going to refactor how the quick switch modal opened so that I could pass in whether it was open as a prop but that turned into bigger can of worms than I wanted to get into and really I'd prefer to come up with a robust way to tell if _any_ modal is open here